### DISCUSSION TITLE
Can't get property container ref element - React-spring v9. ParallaxLayers. useRef

### DISCUSSION THREAD

**Discussion by sampashas** [pmndrs/react-spring#1769] [Support] [answered]
Hello everyone!

I'm not sure if this belongs to the category of questions, because it looks like a bug:

When i want add event scroll to track its position on the page, i used useRef and almost everything worked, except for the smoothness of the scrollToTop event, but that's not about it now. Then I got the value of useRef as an object indicating what kind of object it was and could manipulate it (I needed to get the value of ref.current.container)

Was:
<img width="748" alt="Снимок экрана 2021-11-24 в 15 53 28" src="https://user-images.githubusercontent.com/47352830/143242197-c2a1b133-0a9b-461a-82e0-4f8570e06ff6.png">
console.log(ref)
console.log(ref.current.container)

At that time I used react-spring@8.0.27 with that dependency
`import { Parallax, ParallaxLayer } from 'react-spring/renderprops-addons'`

But when i needed add sticky container it's not worked
And I had to turn to the example from the react-spring documentation

There I encountered other dependencies
`import { Parallax, ParallaxLayer } from '@react-spring/parallax'`
With which the sticky container worked, but the opportunity to get the correct ref was broken

Become: 
object itself disappeared and became impersonal
<img width="696" alt="Снимок экрана 2021-11-24 в 15 52 54" src="https://user-images.githubusercontent.com/47352830/143242337-0940174f-2688-4481-a0fb-28b4242d8c63.png">
console.log(ref)
console.log(ref.current.container)

Question:
How do I get the correct ref in this case?

Thank's for any help!

**Comment by joshuaellis** [comment, marked as answer]
Can you provide a demo of what's not working i the latest version of react-spring please? We're not working on v8 anymore.

**Reply by sampashas** [reply]
Yes, i provide it there, if i understand you correctly


>object itself disappeared and became impersonal
![image](https://user-images.githubusercontent.com/47352830/143258462-95c903ff-d766-438a-8f2c-220f8823830f.png)

This happening when i used react-spring@9.3.0 (@react-spring/parallax). And tried used useRef on Parallax (or ParallaxLayer). 

So now I don't understand how do I get the correct ref in this case, if ref.current.container no longer available. 

Because without this, I can't track scrolling events on my page, as would be the case using the "window.scrollY" analogy, which works only via ref in react-spring

**Reply by joshuaellis** [reply]
so you want to access the dom node they're attached too? I think we've regressed that feature, it doesn't look difficult to add back in. But I want to understand the use-case clearly before deciding to do so. How come you need to access those dom nodes?

**Reply by sampashas** [reply]
I would like to access the reference element in order to be able to manage it in detail, namely to track the scroll position on the page in real time without delay.
Like this when i used scroll before
<img width="331" alt="Снимок экрана 2021-11-25 в 21 58 42" src="https://user-images.githubusercontent.com/47352830/143490742-77a4bbc6-18d6-41af-bbac-570eb5c082e1.png">

If there is any other way, please let me know, but I have already searched a lot of resources and approaches on the codesandbox and, unfortunately, to no avail
Therefore, if you returned this opportunity, it would make my life and other users aftertime very much easier. :)

**Reply by sampashas** [reply]
So, will you help me? or would you can get it back?

**Reply by joshuaellis** [reply]
i've created a feature request – https://github.com/pmndrs/react-spring/issues/1770 as to getting round to implementing it, i'm not sure when that would be. But you're always welcome to contributing it yourself and we can go from there 😄

**Reply by sampashas** [reply]
Great news, thanks! :)
